executor/simple_test.go

```diff
@@ -974,3 +974,17 @@ func (s *testSuite3) TestShowGrantsAfterDropRole(c *C) {
 	tk.MustExec("DROP ROLE r29473")
 	tk.MustQuery("SHOW GRANTS").Check(testkit.Rows("GRANT CREATE USER ON *.* TO 'u29473'@'%'"))
 }
+
+func (s *testSuite3) TestDropRoleAfterRevoke(c *C) {
+	// issue 29781
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test;")
+	tk.Se.Auth(&auth.UserIdentity{Username: "root", Hostname: "localhost"}, nil, nil)
+
+	tk.MustExec("create role r1, r2, r3;")
+	defer tk.MustExec("drop role r2, r3;")
+	tk.MustExec("grant r1,r2,r3 to current_user();")
+	tk.MustExec("set role all;")
+	tk.MustExec("revoke r1, r3 from root;")
```

Comment:
Here `root` will most likely map to `"root"@"%"` which is not the same as `"root"@"localhost"`.

Reply:
I think it may not be a problem.
```
if currentUser == user.Username {
  for i := 0; i < len(activeRoles); i++ {
    if activeRoles[i].Username == role.Username && activeRoles[i].Hostname == role.Hostname {
```
If the user name is the same, we will compare the role's username & hostname.